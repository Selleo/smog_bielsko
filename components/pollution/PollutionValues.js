import React, { Component, PropTypes } from 'react';
import { View, StyleSheet, Text, Dimensions }      from 'react-native';
import header from '../stylesheets/Header'
import ElevatedView from 'react-native-elevated-view'
const window = Dimensions.get('window');

import Icon from '../MyIcons';

export default class PollutionValues extends Component {
  constructor(props) {
    super(props);

    this.state = {
      width: Dimensions.get('window').width,
      height: Dimensions.get('window').height,
    };

    this.onLayout = this.onLayout.bind(this);
  }

  singleComponent(type) {
    return (
      <View style={header.withShadow}>
        <ElevatedView elevation={4} style={header.elevatedView}>
          <View style={{ flexDirection: 'row', justifyContent: 'space-between', padding: 10, alignItems: 'center' }}>
            <Text>{names[type]}</Text>
            <Text style={{fontSize: 18}}>{this.getDatasetItem(type).v} {suffices[type]}</Text>
          </View>
        </ElevatedView>
      </View>
    )
  }

  bigComponent(type) {
    return (
      <View style={{ width: (this.state.width - 60)/ 2, marginHorizontal:5 }}>
        <ElevatedView elevation={4} style={[header.elevatedView, styles.elevatedView]}>
          <View style={[styles.bigComponent]}>
              <View style={{ flexDirection: 'row', justifyContent: 'space-between', flex: 1}}>
                <View style={{ flexDirection: 'column', justifyContent: 'space-between' }}>
                  <Text>{names[type]}</Text>
                  <Text>{this.getDatasetItem(type).v} {suffices[type] }</Text>
                </View>
                <View style={{ alignItems: 'center', justifyContent: 'center' }}>
                  <Text><Icon name={icons[type]} size={25} color="#900" /></Text>
                </View>
              </View>
          </View>
        </ElevatedView>
      </View>
    )
  }

  render() {
    return (
      <View style={{alignItems: 'stretch',}} onLayout={this.onLayout}>
        { this.getDatasetItem('pm10') && this.singleComponent('pm10') }
        { this.getDatasetItem('no2') && this.singleComponent('no2') }
        { this.getDatasetItem('so2') && this.singleComponent('so2') }

        <View style={{ flexDirection: 'row', flexWrap: 'wrap', justifyContent: 'space-between', alignItems: 'flex-start', marginHorizontal: 15 }}>
          { this.getDatasetItem('h') && this.bigComponent('h') }
          { this.getDatasetItem('w') && this.bigComponent('w') }
          { this.getDatasetItem('t') && this.bigComponent('t') }
          { this.getDatasetItem('p') && this.bigComponent('p') }
        </View>

      </View>
    )
  }

  getDatasetItem(type) {
    return this.props.dataset[type]
  }

  onLayout() {
    this.setState({
      width: Dimensions.get('window').width,
      height: Dimensions.get('window').height,
    });
  }

  static propTypes = {
    dataset: PropTypes.object.isRequired
  };
}

const names = {
  'co': 'Tlenek Węgla',
  'h': 'Wilgotność',
  'no2': 'Dwutlenek Azotu',
  'o3': 'Ozon',
  'p': 'Ciśnienie',
  'pm10': 'Pył zawieszony',
  'so2': 'Dwutlenek Siarki',
  't': 'Temperatura',
  'w': 'Wiatr'
};
const icons = {
  't': "thermometer",
  'p': "pressure",
  'h': "rain",
  'w': "wind",

};
const suffices = {
  't': '°C',
  'p': 'hPa',
  'h': '%',
  'w': ' m/s'
};

const styles = StyleSheet.create({
  test: {
    alignItems: 'flex-start',
  },
  bigComponent: {
    alignItems: 'flex-start',
    flex: 1,
    flexWrap: 'wrap',
    paddingHorizontal: 10,
    paddingVertical: 5,
  },
  elevatedView: {
    alignItems: 'center',
    backgroundColor: 'white',
    justifyContent: 'space-between',
    margin: 0,
    marginBottom: 20
  }
});
