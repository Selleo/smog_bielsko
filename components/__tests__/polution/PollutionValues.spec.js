import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import PollutionValues                 from '../../pollution/PollutionValues';

const originalExpect = global.expect;

chai.use(dirtyChai);

let wrapper, props;

describe('PollutionValues', () => {
  beforeEach(() => {
    props = {
      dataset: {
        co: {
          v: 0.1
        },
        h: {
          v: 88.2461
        }
      }
    };

    wrapper = shallow(<PollutionValues {...props}/>);
  });

  it('renders correctly', () => {
    originalExpect(wrapper).toMatchSnapshot();
    expect(wrapper).to.exist();
  });


  it('render component with 3 children', () => {
    expect(wrapper.children().first().children().nodes.join('')).to.equal('Tlenek Węgla: 0.1');
    expect(wrapper.children().last().children().nodes.join('')).to.equal('Wilgotność: 88.2461%');
    expect(wrapper.children().length).to.equal(2)
  });

  describe('#getDatasetKeys', () => {
    it('return keys', () => {
      expect(wrapper.instance().getDatasetKeys()).to.eql(['co', 'h'])
    });
  });

  describe('#getDatasetItem', () => {
    it('return object', () => {
      expect(wrapper.instance().getDatasetItem(0)).to.eql({ v: 0.1 })
    });
  });
});
