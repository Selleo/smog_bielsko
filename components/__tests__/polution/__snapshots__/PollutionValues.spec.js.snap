exports[`PollutionValues renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={Object {}}>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "backgroundColor": "#ccc",
                "color": "#363636",
                "fontSize": 16,
                "paddingHorizontal": 10,
                "paddingVertical": 6,
              }
        }>
        Tlenek Węgla
        : 
        0.1
    </Text>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "backgroundColor": "#ccc",
                "color": "#363636",
                "fontSize": 16,
                "paddingHorizontal": 10,
                "paddingVertical": 6,
              }
        }>
        Wilgotność
        : 
        88.2461
        %
    </Text>
</View>,
  "nodes": Array [
    <View
      style={Object {}}>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "backgroundColor": "#ccc",
                        "color": "#363636",
                        "fontSize": 16,
                        "paddingHorizontal": 10,
                        "paddingVertical": 6,
                      }
            }>
            Tlenek Węgla
            : 
            0.1
      </Text>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "backgroundColor": "#ccc",
                        "color": "#363636",
                        "fontSize": 16,
                        "paddingHorizontal": 10,
                        "paddingVertical": 6,
                      }
            }>
            Wilgotność
            : 
            88.2461
            %
      </Text>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PollutionValues
        dataset={
                Object {
                        "co": Object {
                          "v": 0.1,
                        },
                        "h": Object {
                          "v": 88.2461,
                        },
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PollutionValues {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "dataset": Object {
            "co": Object {
              "v": 0.1,
            },
            "h": Object {
              "v": 88.2461,
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={Object {}}>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "backgroundColor": "#ccc",
                                        "color": "#363636",
                                        "fontSize": 16,
                                        "paddingHorizontal": 10,
                                        "paddingVertical": 6,
                                      }
                    }>
                    Tlenek Węgla
                    : 
                    0.1
          </Text>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "backgroundColor": "#ccc",
                                        "color": "#363636",
                                        "fontSize": 16,
                                        "paddingHorizontal": 10,
                                        "paddingVertical": 6,
                                      }
                    }>
                    Wilgotność
                    : 
                    88.2461
                    %
          </Text>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={Object {}}>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "backgroundColor": "#ccc",
                                        "color": "#363636",
                                        "fontSize": 16,
                                        "paddingHorizontal": 10,
                                        "paddingVertical": 6,
                                      }
                    }>
                    Tlenek Węgla
                    : 
                    0.1
          </Text>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "backgroundColor": "#ccc",
                                        "color": "#363636",
                                        "fontSize": 16,
                                        "paddingHorizontal": 10,
                                        "paddingVertical": 6,
                                      }
                    }>
                    Wilgotność
                    : 
                    88.2461
                    %
          </Text>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PollutionValues
    dataset={
        Object {
            "co": Object {
              "v": 0.1,
            },
            "h": Object {
              "v": 88.2461,
            },
          }
    } />,
}
`;
